s64da_benchmark_toolkit/prepare.py

```diff
@@ -105,21 +116,23 @@ def _run_shell_task(self, task, return_output=False):
 
             if return_output:
                 stdout, _ = p.communicate()
-                return stdout
+                print(f"output of run_shell_task: {stdout.decode('utf-8').strip()}")
+                return stdout.decode('utf-8').strip()
 
     def _run_tasks_parallel(self, tasks):
         def get_future(executor, task):
             if callable(task):
                 return executor.submit(task)
             else:
-                return executor.submit(self._run_shell_task, task)
+                return executor.submit(self._run_shell_task, task, True)
 
         # randomize the tasks to decrease lock contention
         random.shuffle(tasks)
         with ThreadPoolExecutor(max_workers=self.args.max_jobs) as executor:
             futures = [get_future(executor, task) for task in tasks]
             for completed_future in as_completed(futures):
                 exc = completed_future.exception()
+                PrepareBenchmarkFactory.check_ingest(completed_future.result())
                 if exc:
                     print(f'Task threw an exception: {exc}')
```

Comment:
`check_ingest` will raise an error if the ingest failed, so this should be gracefully handled. Meaning you should put it in `try/except NoIngestException` clause. Something like:
```
                exc = completed_future.exception()
                ingest_check_failed = False
                try:
                    PrepareBenchmarkFactory.check_ingest(completed_future.result())
                except NoIngestException:
                    ingest_check_failed = True
                if exc:
                    print(f'Task threw an exception: {exc}')
                    print_tb(exc.__traceback__)
                if exc or ingest_check_failed:
                    for future in futures:
                        future.cancel()
                    exit(1)
```

Reply:
Done